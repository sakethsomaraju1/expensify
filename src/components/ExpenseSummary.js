import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import SelectExpensesTotal from'../selectors/expenses-total'
import {Link} from 'react-router-dom'

export const ExpenseSummary = ({expenseCount,expenseTotal}) =>{
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formatedExpensesTotal = numeral(expenseTotal/100).format('$0,0.00')
    return(
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Veiwing <span>{expenseCount}</span> {expenseWord} totalling <span>{formatedExpensesTotal}</span></h1>
                <div className="page-header__actions">
                <Link to="/create" className="button">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state)=>{
    const visibleExpenses = selectExpenses(state.expenses,state.filters)

    return{
        expenseCount: visibleExpenses.length,
        expenseTotal:SelectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)