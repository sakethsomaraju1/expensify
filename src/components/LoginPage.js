import React from 'react'
import {connect} from 'react-redux'
import {startLogin} from '../actions/auth' 
import {startLogin2} from '../actions/auth'

export const LoginPage = ({startLogin,startLogin2})=>
    (
        <div className="box-layout">
        <h2 className="blacks">Black Lives Matter</h2>
        <div className="box-layout__box">
            <h1 className=".box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control !</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
            <p> </p>
            <button className="button" onClick={startLogin2}>Login with Github</button>

        </div>
        
        <p className="credentials"> Developed by Durga Saketh</p>
        </div>
    )


const mapDispatchToProps = (dispatch) =>({
    startLogin:()=> dispatch(startLogin()),
    startLogin2:()=> dispatch(startLogin2())
    
})

export default connect(undefined,mapDispatchToProps)(LoginPage)