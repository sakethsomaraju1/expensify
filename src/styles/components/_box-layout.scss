.box-layout{
    background:#000017;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;;
    height:100vh;
    width:100vw;
}
.box-layout__box{
    background: fade-out(white, .15);
    border-radius:3px;
    text-align: center;
    width: 25 rem;
    padding: $l-size $m-size;
}
.box-layout__title{
    margin: 0 0 $m-size 0;
    line-height: 1;
}
.credentials{
    color:fade-out(white, .15);
    position: absolute;
    bottom: 10px;
}
.blacks{
    font-size: 33px;
    position: absolute;
    top: 0px;
    color:black !important;
}
