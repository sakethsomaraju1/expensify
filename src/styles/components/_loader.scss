.loader{

    height: 100vh;
    width: 100vh;
    justify-content: center;
    display: flex;
    align-items: center;
}
.loader__image{
    height: 6rem;
    width: 6rem;
}