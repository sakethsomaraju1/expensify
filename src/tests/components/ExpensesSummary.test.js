import React from 'react'
import { shallow } from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary'

test('should render Expense summary with one expense to the dashboard',()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={3000} />)
    expect(wrapper).toMatchSnapshot()
})
test('should render Expense summary with multiple expense to the dashboard',()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={10} expenseTotal={31000}/>)
    expect(wrapper).toMatchSnapshot()
})