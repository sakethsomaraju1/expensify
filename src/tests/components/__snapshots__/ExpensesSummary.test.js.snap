// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense summary with multiple expense to the dashboard 1`] = `
<div>
  <h1>
    Veiwing 
    10
     
    expenses
     totalling 
    $310.00
  </h1>
</div>
`;

exports[`should render Expense summary with one expense to the dashboard 1`] = `
<div>
  <h1>
    Veiwing 
    1
     
    expense
     totalling 
    $30.00
  </h1>
</div>
`;
